mender.go

```diff
@@ -15,28 +15,30 @@ package main
 
 import (
 	"bufio"
-	"encoding/json"
-	"errors"
-	"io/ioutil"
 	"os"
 	"strings"
 	"time"
 
 	"github.com/mendersoftware/log"
+	"github.com/pkg/errors"
 )
 
 type Controler interface {
-	Bootstrap() error
-	TransitionState() MenderState
+	Bootstrap() menderError
 	GetCurrentImageID() string
-	GetDaemonConfig() daemonConfig
-	GetUpdaterConfig() httpsClientConfig
-	LastError() error
+	GetUpdatePollInterval() time.Duration
+	HasUpgrade() (bool, menderError)
+	CheckUpdate() (*UpdateResponse, menderError)
```

Comment:
Doesn't it have to anyway? And in fact I see this bit differently. I think that controller should be main point where you are controlling the states and it should delegate the work to all needed sub-modules instead of `states` implementing controller interface.
The drawback of leaving this as is looks to be the fact that this interface will grow a lot once we will add new functionalities (like reporting, inventory and so on).
But this is just different view and something we should decide somewhere else (aka some meeting where we will go through architecture and discuss it further). I am OK with leaving this as is.